src/main/java/duke/handler/Handlers.java

```diff
@@ -0,0 +1,265 @@
+package duke.handler;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.LinkedList;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Tasklist;
+import duke.task.Todo;
+
+/**
+ * Enum class used to handle the different commands used in the application.
+ */
+public enum Handlers {
+    Bye("bye"),
+    Deadline("deadline"),
+    Delete("delete"),
+    Event("event"),
+    Find("find"),
+    Mark("mark"),
+    List("list"),
+    Todo("todo"),
+    Unmark("unmark");
+
+    // Only date is provided in input array
+    private static final int DATE = 3;
+    // Both date and time is provided in input array
+    private static final int DATEANDTIME = 4;
+
+    public final String label;
+
+    /**
+     * Enum constructor to reference the exact commands used.
+     *
+     * @param label Exact commands used in the application.
+     */
+    Handlers(String label) {
+        this.label = label;
+    }
+
+    /**
+     * Handles and distributes all commands to the respective handlers.
+     *
+     * @param list Tasklist that contains all tasks.
+     * @param input Text that the user has entered.
+     */
+    public static String commandHandler(Tasklist list, String input) {
+        String[] inputArr = input.split(" ");
+        String cmd = inputArr[0];
+        try {
+            if (cmd.equals(Bye.label)) {
+                return Handlers.byeHandler(list);
+            } else if (cmd.equals(List.label)) {
+                return Handlers.listHandler(list);
+            } else if (cmd.equals(Deadline.label)) {
+                return Handlers.deadlineHandler(list, input, cmd);
+            } else if (cmd.equals(Event.label)) {
+                return Handlers.eventHandler(list, input, cmd);
+            } else if (cmd.equals(Todo.label)) {
+                return Handlers.todoHandler(list, input);
+            } else if (cmd.equals(Mark.label)) {
+                return Handlers.markHandler(list, input);
+            } else if (cmd.equals(Unmark.label)) {
+                return Handlers.unmarkHandler(list, input);
+            } else if (cmd.equals(Find.label)) {
+                return Handlers.findHandler(list, input);
+            } else if (cmd.equals(Delete.label)) {
+                return Handlers.deleteHandler(list, input);
+            } else {
+                throw new IllegalArgumentException("Sorry, this is not a recognized command.\n");
```

Comment:
——— For PR review ———
Consider looping through the enums instead, for less verbosity and better code readability.
ie. something like:
```
for (Handler cmdType : Handler.values()) {
    if(cmd.equals(cmdType.label))
        return [Handler based on cmdType]
    ...
```
This impl. requires no if-statement addition whenever a new command is added.
Some code restructuring here is required to achieve it but it should be worth it.